### ISSUE TITLE
Straight through estimator (STE) for LDPC5GEncoder

### ISSUE THREAD

**Issue by csankar69** [NVlabs/sionna#699]
I am not sure why the following STE inside a class definition is not backproping gradients - I get a 'Gradients do not exist' warning. What is the correct way to pass gradients through an LDPC encoder?
```import tensorflow as tf
class model(Model):
  def __init__(..):
    ..

  @tf.custom_gradient
  def steLdpcEncoder(self, x):
    y = self.ldpcEncoder(tf.round(x))
    def grad(dy):
      return dy
    return y, grad
  
  def call(...):
    x = ...
    encodedBits = self.steLdpcEncoder(x)

**Comment by SebastianCa** [comment]
Hi @csankar69,
Without knowing the details of your experiment, it’s hard to give specific advice. Could you provide a minimal example that reproduces the issue? I quickly tested accessing the gradients, and it seems to work in my setup.

What’s the motivation for using an STE for the encoder? Conceptually, I’m not convinced this produces meaningful gradients (encoding is binary i.e. XOR operations). For instance, consider a toy example where the encoder only interleaves the input without adding redundancy. In such a case, your gradients would completely ignore this interleaving step.

I'll transfer this thread to discussions as it is not an issue with Sionna.
